sdb/commands/zfs/vdev.py

```diff
@@ -87,6 +87,11 @@ def pretty_print(self,
                 level += 2
                 pvd = pvd.vdev_parent
 
+            if vdev.vdev_isl2cache != 0:
+                print("".ljust(indent), "-".ljust(18), "-".ljust(7),
+                      "-".ljust(4), "".ljust(level), "cache")
+                level = 2
```

Comment:
I may be wrong but won't this print the `- - - cache` header line above each cache device entry?

In MDB this would be printed once:
```
> ::spa -v
ADDR                 STATE NAME
ffffff03afc1b000    ACTIVE meta-domain

    ADDR             STATE     AUX          DESCRIPTION
    ffffff0385108800 HEALTHY   -            root
    ffffff03850ba800 HEALTHY   -              raidz
    ffffff0384fd7000 HEALTHY   -                /tmp/dks0
    ffffff0384fee000 HEALTHY   -                /tmp/dks1
    ffffff0385101800 HEALTHY   -                /tmp/dks2
    -                -         -            cache
    ffffff037a48c800 HEALTHY   -              /dev/dsk/c2t1d0s0
    ffffff03850c7800 HEALTHY   -              /dev/dsk/c2t2d0s0
    -                -         -            spares
    ffffff038269a000 HEALTHY   -              /dev/dsk/c2t3d0s0
ffffff0375690000    ACTIVE rpool

    ffffff037571b000 HEALTHY   -            root
    ffffff03758c7800 HEALTHY   -              /dev/dsk/c2t0d0s0
```

Reply:
Dang!  That's true.

Reply:
Ok.  Let me know if there was a better way to code it up.   Here's the output.
```
sdb> spa -v
ADDR               NAME
------------------------------------------------------------
0xffff88ff177f4000 tank
      ADDR               STATE   AUX  DESCRIPTION
      ------------------------------------------------------------
      0xffff88ff1d788000 HEALTHY NONE  root
      0xffff88ff1d7a0000 HEALTHY NONE    /root/file1
      0xffff88ff1d6cc000 HEALTHY NONE    /root/file4
      -                  -       -     cache
      0xffff88ff1d628000 HEALTHY NONE    /root/file2
      0xffff88ff1d3ac000 HEALTHY NONE    /root/file5
sdb> 
```

Reply:
In my pool "tank", /root/file4 is a log device, so I think had better add code for the log keyword preceding the vdev just like for l2arc vdev(s).
```
root@ub-20:~/sdb# zpool status
  pool: tank
 state: ONLINE
config:

	NAME           STATE     READ WRITE CKSUM
	tank           ONLINE       0     0     0
	  /root/file1  ONLINE       0     0     0
	logs	
	  /root/file4  ONLINE       0     0     0
	cache
	  /root/file2  ONLINE       0     0     0
	  /root/file5  ONLINE       0     0     0
	spares
	  /root/file3  AVAIL   
```

Reply:
Added code for log and spares.
```
sdb> spa -v
ADDR               NAME
------------------------------------------------------------
0xffff88ff1d718000 tank
      ADDR               STATE   AUX  DESCRIPTION
      ------------------------------------------------------------
      0xffff88ff297d0000 HEALTHY NONE  root
      0xffff88ff17014000 HEALTHY NONE    /root/file1
      -                  -       -     logs
      0xffff88ff1212c000 HEALTHY NONE    mirror
      0xffff88ff1d570000 HEALTHY NONE      /root/file4
      0xffff88ff1d548000 HEALTHY NONE      /root/file6
      -                  -       -     cache
      0xffff88ff1d550000 HEALTHY NONE    /root/file2
      0xffff88ff1d5d8000 HEALTHY NONE    /root/file5
      -                  -       -     spares
      0xffff88ff297cc000 HEALTHY NONE    /root/file3
sdb> 
root@ub-20:~/sdb# zpool status
  pool: tank
 state: ONLINE
config:

	NAME             STATE     READ WRITE CKSUM
	tank             ONLINE       0     0     0
	  /root/file1    ONLINE       0     0     0
	logs	
	  mirror-1       ONLINE       0     0     0
	    /root/file4  ONLINE       0     0     0
	    /root/file6  ONLINE       0     0     0
	cache
	  /root/file2    ONLINE       0     0     0
	  /root/file5    ONLINE       0     0     0
	spares
	  /root/file3    AVAIL   

errors: No known data errors
root@ub-20:~/sdb# 
```